### PR TITLE
Update kubevirtci

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

Updates to latest `kubevirtci` to fix k8s-1.17 so that the lane works finally.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
/lgtm

can we configure the new lane as not-reporting until we have sorted out the remaining issues?

**Comment by dhiller** [comment]
> /lgtm
> 
> can we configure the new lane as not-reporting until we have sorted out the remaining issues?

Please see kubevirt/project-infra#331

**Comment by dhiller** [comment]
Updated the lane k8s-1.17 to not reporting and always run. So we should be good here.

**Comment by danielBelenky** [comment]
@dhiller is it still a draft intentionally?

**Comment by dhiller** [comment]
@danielBelenky No, I think we agreed to get this merged and collect data. AFAIU everything is setup according to to that now.

**Comment by dhiller** [comment]
Still got this weird sriov lane issue though. @phoracek IDK what to do here, can you have a look when you have time, please?

**Comment by rmohr** [comment]
@phoracek maybe here we see the issue which was reported in #3035 ?

**Comment by phoracek** [comment]
It did not even get to do anything with SR-IOV, it failed due to the deprecated `extensions` group.

```
Not sure what to do next? 😅 Check out https://kind.sigs.k8s.io/docs/user/quick-start/
ipv6 cni: 
no ipv6, safe to install flannel
clusterrole.rbac.authorization.k8s.io/flannel created
clusterrolebinding.rbac.authorization.k8s.io/flannel created
serviceaccount/flannel created
configmap/kube-flannel-cfg created
unable to recognize "/home/prow/go/src/github.com/kubevirt/kubevirt/cluster-up//cluster/kind/manifests/kube-flannel.yaml": no matches for kind "PodSecurityPolicy" in version "extensions/v1beta1"
unable to recognize "/home/prow/go/src/github.com/kubevirt/kubevirt/cluster-up//cluster/kind/manifests/kube-flannel.yaml": no matches for kind "DaemonSet" in version "extensions/v1beta1"
unable to recognize "/home/prow/go/src/github.com/kubevirt/kubevirt/cluster-up//cluster/kind/manifests/kube-flannel.yaml": no matches for kind "DaemonSet" in version "extensions/v1beta1"
unable to recognize "/home/prow/go/src/github.com/kubevirt/kubevirt/cluster-up//cluster/kind/manifests/kube-flannel.yaml": no matches for kind "DaemonSet" in version "extensions/v1beta1"
unable to recognize "/home/prow/go/src/github.com/kubevirt/kubevirt/cluster-up//cluster/kind/manifests/kube-flannel.yaml": no matches for kind "DaemonSet" in version "extensions/v1beta1"
unable to recognize "/home/prow/go/src/github.com/kubevirt/kubevirt/cluster-up//cluster/kind/manifests/kube-flannel.yaml": no matches for kind "DaemonSet" in version "extensions/v1beta1"
Makefile:92: recipe for target 'cluster-up' failed
```

**Comment by rmohr** [comment]
@phoracek looks like the kind updates are only half done then. I would suggest to update the used manifests ...

**Comment by rmohr** [comment]
> We will look into it and fix the Flannel manifest.

Thx.

**Comment by dhiller** [comment]
@phoracek maybe you can ping me when you've done the changes and I'll incorporate them here so that we don't destroy the sriov lane by merging this one?

/hold

**Comment by phoracek** [comment]
@dhiller will do. Thanks for the patience.

**Comment by dhiller** [comment]
@qinqon just updated the kubevirtci

/test pull-kubevirt-e2e-k8s-1.17.0

**Comment by dhiller** [comment]
@qinqon FTR - test results should be here when finished: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/3046/pull-kubevirt-e2e-k8s-1.17.0/1227980856329506816

**Comment by danielBelenky** [comment]
@dhiller thanks for this update :)
/lgtm

Not merging for now, waiting for @phoracek / @qinqon to approve

**Comment by phoracek** [comment]
SR-IOV lane is still failing due to extensions API. Or Mergi is addressing that at the moment.

**Comment by dhiller** [comment]
@qinqon @phoracek new test results here: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/3046/pull-kubevirt-e2e-k8s-1.17.0/1227980856329506816

Still 28 tests failing. Maybe some update missing somewhere?

ICYMI: With the changes in place you also can directly call `functest` kubevirt using a cluster that was locally provisioned before using kubevirtci. Did you folks try that already? This could make your iterations way faster.

In a nutshell:
- in `kubevirtci` provision cluster locally `cd cluster-provision/k8s/1.17.0 && ./provision.sh`
- `export KUBEVIRTCI_PROVISION_CHECK=1`
- copy local changes from `kubevirtci/cluster-up` to `kubevirt/cluster-up`
- `make cluster-up && make cluster-sync`
- `make functest` (optionally use ginkgo focus, i.e. `FUNC_TEST_ARGS='-ginkgo.focus=vmi_cloudinit_test -ginkgo.regexScansFilePath' make functest`)

**Comment by dhiller** [comment]
Created a PR to document this: kubevirt/kubevirtci#284

**Comment by qinqon** [comment]
@dhiller problem with masquerade tests is containers not having ip forwarding, i am preparing a fix at kubevirtci level, but we have to think about a kubevirt proper fix.
